packages/api/src/command/medical/cq-directory/create-cq-directory-entry.ts

```diff
@@ -81,34 +78,35 @@ export const createOrUpdateCQDirectoryEntries = async (
 
 async function computeEarthPoint(orgData: CQDirectoryEntryData): Promise<string | undefined> {
   if (orgData.lat && orgData.lon) {
-    const query = "SELECT ll_to_earth(:lat, :lon) as point";
-    const pointQueryResult = await sequelize.query(query, {
-      replacements: { lat: orgData.lat, lon: orgData.lon },
-      type: QueryTypes.SELECT,
-      logging: false,
+    const pointQueryResult = await CQDirectoryEntryModel.findOne({
+      attributes: [[sequelize.fn("ll_to_earth", orgData.lat, orgData.lon), "point"]],
+      raw: true,
     });
     const point = QueryResultSchema.parse(pointQueryResult);
-    return point[0].point;
+    return point.point;
   }
   return;
 }
 
 const createCQDirectoryEntries = async (orgDataArray: CQDirectoryEntryData[]): Promise<void> => {
-  for (const orgData of orgDataArray) {
-    const point = await computeEarthPoint(orgData);
-    orgData.point = point;
-  }
+  await Promise.all(
+    orgDataArray.map(async orgData => {
+      orgData.point = await computeEarthPoint(orgData);
+    })
+  );
 
   await CQDirectoryEntryModel.bulkCreate(orgDataArray);
 };
 
 const updateCQDirectoryEntries = async (
   updateEntries: CQDirectoryEntryDataWithUpdateAndId[]
 ): Promise<void> => {
-  for (const entry of updateEntries) {
-    await CQDirectoryEntryModel.update(entry, {
-      where: { id: entry.id },
-      returning: true,
-    });
-  }
+  await Promise.allSettled(
```

Comment:
just checking that `promise.all` is desired behavior in createCQDirectoryEntries and `promise.allSettled` desired in updateCQDirectoryEntries

Reply:
hmm.. so like having it fail in case any one org fails be added to the table?